handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslContext.java

```diff
@@ -0,0 +1,782 @@
+    private final long sessionCacheSize;
+    private final long sessionTimeout;
+    private final OpenSslApplicationProtocolNegotiator apn;
+    private final int mode;
+
+    // Reference Counting
+    private volatile int refCnt = 1;
+    private final ResourceLeak leak;
+
+    final Certificate[] keyCertChain;
+    final ClientAuth clientAuth;
+    final OpenSslEngineMap engineMap = new DefaultOpenSslEngineMap();
+    volatile boolean rejectRemoteInitiatedRenegotiation;
+
+    static final OpenSslApplicationProtocolNegotiator NONE_PROTOCOL_NEGOTIATOR =
+            new OpenSslApplicationProtocolNegotiator() {
+                @Override
+                public ApplicationProtocolConfig.Protocol protocol() {
+                    return ApplicationProtocolConfig.Protocol.NONE;
+                }
+
+                @Override
+                public List<String> protocols() {
+                    return Collections.emptyList();
+                }
+
+                @Override
+                public ApplicationProtocolConfig.SelectorFailureBehavior selectorFailureBehavior() {
+                    return ApplicationProtocolConfig.SelectorFailureBehavior.CHOOSE_MY_LAST_PROTOCOL;
+                }
+
+                @Override
+                public ApplicationProtocolConfig.SelectedListenerFailureBehavior selectedListenerFailureBehavior() {
+                    return ApplicationProtocolConfig.SelectedListenerFailureBehavior.ACCEPT;
+                }
+            };
+
+    static {
+        AtomicIntegerFieldUpdater<ReleasableOpenSslContext> updater =
+                PlatformDependent.newAtomicIntegerFieldUpdater(ReleasableOpenSslContext.class, "refCnt");
+        if (updater == null) {
+            updater = AtomicIntegerFieldUpdater.newUpdater(ReleasableOpenSslContext.class, "refCnt");
+        }
+        refCntUpdater = updater;
+
+        List<String> ciphers = new ArrayList<String>();
+        // XXX: Make sure to sync this list with JdkSslEngineFactory.
+        Collections.addAll(
+                ciphers,
+                "ECDHE-RSA-AES128-GCM-SHA256",
+                "ECDHE-RSA-AES128-SHA",
+                "ECDHE-RSA-AES256-SHA",
+                "AES128-GCM-SHA256",
+                "AES128-SHA",
+                "AES256-SHA",
+                "DES-CBC3-SHA");
+        DEFAULT_CIPHERS = Collections.unmodifiableList(ciphers);
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("Default cipher suite (OpenSSL): " + ciphers);
+        }
+
+        Integer dhLen = null;
+
+        try {
+            String dhKeySize = AccessController.doPrivileged(new PrivilegedAction<String>() {
+                @Override
+                public String run() {
+                    return SystemPropertyUtil.get("jdk.tls.ephemeralDHKeySize");
+                }
+            });
+            if (dhKeySize != null) {
+                try {
+                    dhLen = Integer.parseInt(dhKeySize);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method io.netty.handler.ssl.ReleasableOpenSslContext.<static initializer for ReleasableOpenSslContext>() converts String to boxed primitive using excessive boxing [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=fb-contrib%3ANAB_NEEDLESS_BOXING_VALUEOF)